### PR TITLE
add dtype for nn.functional.softmax

### PR DESCRIPTION
**Pull request by ofhwei**

### CODE REVIEW

File: python/oneflow/nn/functional/softmax.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+limitations under the License.
+"""
+import os
+from typing import List, Optional
+
+from oneflow.framework.tensor import Tensor
+import oneflow as flow
+
+# ref https://github.com/pytorch/pytorch/blob/master/torch/nn/functional.py
+def softmax(input: Tensor, dim: Optional[int] = None, dtype = None) -> Tensor:
```
Comment (daquexian) [inline python/oneflow/nn/functional/softmax.py:23]: 可以添加一个对 dtype 参数的单元测试

**Review by daquexian — APPROVED** [review]

File: python/oneflow/nn/functional/softmax.py (hunk 1 above)
Comment (ofhwei) [reply python/oneflow/nn/functional/softmax.py:23]: 👌

**Review by liujuncheng — APPROVED** [review]
